File: LeetCode/205. Isomorphic Strings/Solution.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        d = {}
        d2 = {}

        for i in range(0, len(s), 1):
            if s[i] not in d and t[i] not in d2:
                d[s[i]] = t[i]
                d2[t[i]] = s[i]
            elif s[i] in d and t[i] in d2:
                if d[s[i]] != t[i] or d2[t[i]] != s[i]: return False
            else: return False
        return True

        

File: LeetCode/205. Isomorphic Strings/Solution.cs
public class Solution {
    public bool IsIsomorphic(string s, string t) {
        var dict = new Dictionary<char, char>();
        var dict2 = new Dictionary<char, char>();
        var l = s.Length;

        for(var i = 0; i < l; i++){
            if(!dict.ContainsKey(s[i]) && !dict2.ContainsKey(t[i])){
                dict[s[i]] = t[i];
                dict2[t[i]] = s[i];
            }else if(dict.ContainsKey(s[i]) && dict2.ContainsKey(t[i])){
                if(dict[s[i]] != t[i] || dict2[t[i]] != s[i]) return false;
            }
            else return false;
        }
        return true;
    }
}